libiconv-full: update to version 1.16

BKPepe opened the pull request:
This patch was sent to openwrt-devel mailing list, but since 20th July (almost a month and nobody cares about it) is stuck in the moderator queue. 
```
Your mail to 'openwrt-devel' with the subject

    [PATCH] libiconv-full: update to version 1.16

Is being held until the list moderator can review it for approval.

The reason it is being held:

    Message body is too big: 1167419 bytes with a limit of 512 KB

Either the message will get posted to the list, or you will receive
notification of the moderator's decision.  If you would like to cancel
this posting, please visit the following URL:

-------
```



- Removed following patches:
100-strip_charsets.patch - makes the full variant slim.
101-autotools.patch - this one fails to apply because it was backported
from newer versions for 1.11.1.
103-configure_ac_fix.patch - backported from newer versions
200-work-with-libtool2.patch - is not needed anymore, it is done
differently in upstream
300-fortify-source-compat.patch - these files are not there anymore

- TVHeadend requires working iconv library e.g. transliteration to ASCII
and this does not work with libiconv-full currently.

There is a simple test, which requires to install iconv package.

Before applying this update:
root@turris:/# echo ŽluťoučkýKůň | iconv -t ASCII//TRANSLIT//IGNORE
luoukK

After applying this update:
root@turris:~# echo ŽluťoučkýKůň | iconv -t ASCII//TRANSLIT//IGNORE
Zlutouck'yKun

- Makefile changes:
Use HTTPS for their website
Move PKG_MAINTAINER above PKG_LICENSE

Compile and run tested on Turris 1.1, powerpc8540, Turris Omnia, mvebu cortex-a9

BKPepe committed:
libiconv-full: update to version 1.16

- Removed following patches:
100-strip_charsets.patch - makes the full variant slim.
101-autotools.patch - this one fails to apply because it was backported
from newer versions for 1.11.1.
103-configure_ac_fix.patch - backported from newer versions
200-work-with-libtool2.patch - is not needed anymore, it is done
differently in upstream
300-fortify-source-compat.patch - these files are not there anymore

- TVHeadend requires working iconv library e.g. transliteration to ASCII
and this does not work with libiconv-full currently.

There is a simple test, which requires to install iconv package.

Before applying this update:
root@turris:/# echo ŽluťoučkýKůň | iconv -t ASCII//TRANSLIT//IGNORE
luoukK

After applying this update:
root@turris:~# echo ŽluťoučkýKůň | iconv -t ASCII//TRANSLIT//IGNORE
Zlutouck'yKun

- Makefile changes:
Use HTTPS for their website
Fixed deprecated SPDX License Identifier
Move PKG_MAINTAINER above PKG_LICENSE

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>
Tested-by: Rosen Penev <rosenp@gmail.com> [malta]

neheb commented on package/libs/libiconv-full/Makefile:
```diff
@@ -8,15 +8,15 @@
 PKG_BUILD_DIR:=$(BUILD_DIR)/libiconv-$(PKG_VERSION)
+
+PKG_MAINTAINER:=Jo-Philipp Wich <jo@mein.io>
 PKG_LICENSE:=LGPL-2.0
```
Should be LGPL-2.1-or-later

BKPepe replied:
While touching it, yeah, I could change it. I will do it.

neheb commented:
Tested this locally on malta. Works fine.

BKPepe commented:
> Tested this locally on malta. Works fine.

Just to be sure. Does that mean that I can use your `Tested-by:`?

neheb commented:
Go ahead.

SqTER-PL commented:
I've been using this version for over a year.  Everything works fine.

neheb commented:
I recommend this cleanup:

``
--- a/package/libs/libiconv-full/Makefile
+++ b/package/libs/libiconv-full/Makefile
@@ -21,6 +21,8 @@ PKG_LICENSE:=LGPL-2.1-or-later
 PKG_LICENSE_FILES:=COPYING.LIB
 
 PKG_FIXUP:=patch-libtool
+PKG_INSTALL:=1
+PKG_BUILD_PARALLEL:=1
 
 include $(INCLUDE_DIR)/package.mk
 
@@ -53,21 +55,12 @@ define Package/iconv
   TITLE+= utility
 endef
 
-TARGET_CFLAGS += $(FPIC) -DUSE_DOS -std=gnu89
-
 CONFIGURE_ARGS += \
 	--enable-shared \
 	--enable-static \
 	--disable-rpath \
 	--enable-relocatable
 
-define Build/Compile
-	$(MAKE) -C $(PKG_BUILD_DIR) \
-		CC="$(TARGET_CC)" \
-		DESTDIR="$(PKG_INSTALL_DIR)" \
-		install
-endef
-
 define Build/InstallDev
 	$(INSTALL_DIR) $(1)/usr/lib/libiconv-full/include
 	$(CP) $(PKG_INSTALL_DIR)/usr/include/iconv.h $(1)/usr/lib/libiconv-full/include/`
```

BKPepe commented:
@neheb What would you say if you will use Markdown? To be honest, right now, it looks like garbage. Anyway, you are adding PKG_BUILD_PARALLEL and PKG_INSTALL among other things. Even though, it is cleanup, did you do any test?

neheb commented:
Was missing `. Anyway, I did. I amended that change to my branch.

neheb committed:
libiconv-full: Makefile polishing

- Added PKG_INSTALL to avoid use define Build/Compile
- Added PKG_BUILD_PARALLEL for faster compilation
- Removed TARGET_CLAFGS, it is not longer necessary.
gnu89 is superseded by gnu99, anyway gnu99 was using with GCC 4.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>
[Added commit message]

BKPepe commented:
The final package is even smaller with your changes, @neheb! Thank you.
Because those are your changes, I did it in a separate commit. Hopefully, you don't mind that, because I don't want to be author of something, which I didn't do. I would appreciate if you can check my commit message.

neheb commented:
It's smaller because -DUSE_DOS is no longer set. I have no idea why it was originally set.

adschm commented:
2nd commit seems to lack @neheb 's Signed-off-by.

BKPepe commented:
Done.

aparcar commented:
Thanks, merged.
